src/BtcProofUtils.sol

```diff
@@ -231,7 +232,6 @@ library BtcProofUtils {
             offset += 4;
             uint256 nInScriptBytes;
             (nInScriptBytes, offset) = readVarInt(rawTx, offset);
-            require(nInScriptBytes <= 32, "Scripts over 32 bytes unsupported");
```

Comment:
one more thing we could do.

so a bit of digging shows that scriptSig (output script) should always be < 32 bytes for all standard bitcoin transaction types

but scriptPubKey (input script) can come from anywhere & can be longer

you could remove `bytes32 script` entirely from `TxInput` for some gas savings.

bitcoin mirror users want to prove transactions pay certain people (that's the main use cases) and *maybe* that a transaction spends a previous utxo (BitcoinTxIn.prevTxId) but i can't think of any reason a user would ever need to prove things about the `scriptPubKey` (and if they do, they can use offset+length information this parser gives them to read it themselves directly)